### PR TITLE
[custom channels]: Use new `tapchannelrpc` RPC methods

### PR DESCRIPTION
**Pull request by guggero**
Depends on https://github.com/lightninglabs/taproot-assets/pull/1048.
~Depends on https://github.com/lightningnetwork/lnd/pull/8947.~

Uses the new `tapchannelrpc` RPC methods `SendPayment` and `AddInvoice` that do the RFQ part inline, so we don't have to keep that logic in the `litcli` code.

### CODE REVIEW

**Comment by guggero** [comment]
I tested all commands manually, everything seems to work as expected. Ready for full review now.

File: accounts/service_test.go [hunk 1]
```diff
@@ -100,23 +100,69 @@ func (m *mockLnd) assertInvoiceRequest(t *testing.T, addIndex,
...
+	if m.invoiceSubscriptionErr != nil {
+		return nil, nil, m.invoiceSubscriptionErr
+	}
+
+	m.invoiceReq <- req
+
+	return m.invoiceChan, m.invoiceErrChan, nil
+}
+
+type mockRouter struct {
```
Comment (ffranr) [inline accounts/service_test.go:118]: I think we should add a bit more to this commit message. I'm not certain what's happening here. I think you're removing functionality from the LND mock and putting it into a new mock called `mockRouter`.

Is that split related to the lndclient version bump?

File: cmd/litcli/ln.go [hunk 2]
```diff
@@ -223,8 +224,61 @@ var (
...
+		Usage: "the amount of the asset to send in the asset keysend " +
+			"payment",
+	}
+
+	rfqPeerPubKeyFlag = cli.StringFlag{
+		Name: "rfq_peer_pubkey",
+		Usage: "(optional) the public key of the peer to ask for a " +
+			"quote when converting from assets to sats; must be " +
+			"set if there are multiple channels with the same " +
+			"asset ID present",
```
Comment (ffranr) [inline cmd/litcli/ln.go:177]: Currently, this must be set before it reaches the tap RFQ service. If this field is optional, and unset by the user, when does it get populated?

File: cmd/litcli/ln.go [hunk 3]
```diff
@@ -223,8 +224,61 @@ var (
...
+	}
 )
 
+type resultStreamWrapper struct {
```
Comment (ffranr) [inline cmd/litcli/ln.go:181]: missing doc for struct, struct fields, and `Recv` method

File: cmd/litcli/ln.go [hunk 4]
```diff
@@ -339,15 +360,20 @@ func sendPayment(ctx *cli.Context) error {
 			"is instead: %v", len(destNode))
 	}
 
+	rfqPeerKey, err := hex.DecodeString(ctx.String(rfqPeerPubKeyFlag.Name))
+	if err != nil {
+		return fmt.Errorf("unable to decode RFQ peer public key: "+
+			"%w", err)
+	}
```
Comment (ffranr) [inline cmd/litcli/ln.go:305]: Does this work if the field is optionally unset?

File: itest/assets_test.go [hunk 5]
```diff
@@ -584,31 +584,42 @@ func sendAssetKeySendPayment(t *testing.T, src, dst *HarnessNode, amt uint64,
...
+		PaymentRequest: sendReq,
+	})
 	require.NoError(t, err)
 
-	htlcCarrierAmt := btcAmt.UnwrapOr(500)
-	sendKeySendPayment(
-		t, src, dst, btcutil.Amount(htlcCarrierAmt),
-		encodeResp.CustomRecords,
-	)
+	time.Sleep(time.Second)
```
Comment (ffranr) [inline itest/assets_test.go:615]: Unnecessary sleep? Shouldn't `getAssetPaymentResult` block on the recv?

File: accounts/service_test.go (hunk 1 above)
Comment (GeorgeTsagk) [reply accounts/service_test.go:118]: +1, fully understand the diff but missing rationale/context

File: cmd/litcli/ln.go (hunk 2 above)
Comment (GeorgeTsagk) [reply cmd/litcli/ln.go:177]: Tapd is able to find a peer pubkey, given a specific asset id (it will look for a peer that has that asset channel)

File: cmd/litcli/ln.go (hunk 4 above)
Comment (GeorgeTsagk) [reply cmd/litcli/ln.go:305]: yes, decoding an empty string ("") is valid and doesn't return an error

File: itest/assets_test.go (hunk 5 above)
Comment (guggero) [reply itest/assets_test.go]: Unfortunately and strangely the tests fail without it... It shouldn't make a difference, but somehow it does. Moved it to a single place and added an explanatory comment.

File: accounts/service_test.go (hunk 1 above)
Comment (guggero) [reply accounts/service_test.go:118]: Added some rationale to the commit message.

**Review by ffranr — APPROVED** [review]
